compiler-core/src/type_/error.rs

```diff
@@ -873,11 +894,11 @@ impl FeatureKind {
             FeatureKind::ConstantStringConcatenation => Version::new(1, 4, 0),
             FeatureKind::UnannotatedUtf8StringSegment => Version::new(1, 5, 0),
             FeatureKind::RecordUpdateVariantInference
-            | FeatureKind::RecordAccessVariantInference => Version::new(1, 6, 0),
+            | FeatureKind::RecordAccessVariantInference
+            | FeatureKind::VariantWithDeprecatedAnnotation => Version::new(1, 6, 0),
```

Comment:
Since this will appear in 1.7, could you update `VariantWithDeprecatedAnnotation` to require version `Version::new(1, 7, 0)`?